### ISSUE TITLE
JoinWith with AS in table name generates invalid SQL

### ISSUE THREAD

**Issue by skolodyazhnyy** [go-rel/sql#54]
After updating to latest rel version `rel.JoinWith` with "AS" bit in the table name and filters causes builder to produce invalid SQL. 

Here is an example:

```go
s.orm.Count(ctx, "table2",
	rel.JoinWith(
		"LEFT JOIN",
		"table1 as x",
		"table2.id",
		"x.row_id",
		rel.Eq("x.column_id", "1"),
	),
)
```

Generates SQL query:

```sql
SELECT count(*) AS result FROM `table2` 
  LEFT JOIN `table1` AS `x` ON `table2`.`id`=`x`.`row_id` AND `table1` AS `x`=?;
```

Such SQL produces an error:

```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AS `x`=?' at line 1
```

I traced it down to `builder/query.go` but can't quite figure out exact change which broke code in my example.

**Comment by lihka1202** [comment]
Hi!

I don't know how the Go side of things works as I haven't looked very deeply into the codebase, but based on the SQL Query alone, I noticed a few issues.

1. `AS` has been used twice on the same table in the same query. Ideally, they should be unique.
2. I'm not sure, but I think using `LEFT JOIN` AND `AND` in the same clause is invalid or prone to many errors. 
3. When using `=?`, I think this would need to be used on a specific column and not the whole table.

So a model query could look something like this instead:
```SQL
SELECT COUNT(*) AS result FROM `table2`
LEFT JOIN `table1` AS `x` ON `table2`.`id` = `x`.`row_id`
WHERE `x`.`some_column` = ?; 
```

Based on this, perhaps it's possible to reverse-engineer the problem.

Then again, I'm very new to SQL and Go, so any and all insights are appreciated!

Thank You!

**Comment by skolodyazhnyy** [comment]
This is **useless** @lihka1202, thanks.

**Comment by skolodyazhnyy** [comment]
@lihka1202 this repository is part of Go ORM, a library which helps to convert data between a relational database and Go data structures. 

In this PR I've reported that SQL queries generated by the library are invalid: instead of placing column name `column_id = ?`, the library inserted table name with AS. So, yes, the query is invalid (your #1 and #3), this is the point of this issue.

There is no problem with using AND in ON portion of JOIN operation.

**Comment by skolodyazhnyy** [comment]
Meanwhile, @Fs02 do you have some suggestion about the fix? I see `Buffer.escape` is being used for different "use-cases" so I'm not sure if it makes sense to separate "table name escaping" and "field name escaping" or what. Do you have an opinion?

**Comment by lafriks** [comment]
I have created fix in PR #58

**Comment by Fs02** [comment]
Thanks @lafriks 

@skolodyazhnyy could you test if master have fixed your issue? thanks
